low-level/files/opt/seagate/sspl/setup/resource_map/server.py

```diff
@@ -234,4 +234,38 @@ def prepare_mem_json(self, status, description):
         self.set_health_data(
             memory_dict, status=status, description=description,
             specifics=specifics)
-        return memory_dict
\ No newline at end of file
+        return memory_dict
+
+    def get_fanmodules_info(self):
+        """Get the Fan sensor information using ipmitool."""
+        data = []
+        sensor_reading = self._ipmi.get_sensor_list_by_type('Fan')
```

Comment:
I think this `ipmi` function call should be in try catch block. Although I am not sure what specific exceptions it could throw.

Reply:
@jjxsg  ipmi_client is logging an error in case of failure and returning None,
so no need to add try except block here.
But yes, we can add a check,
 if response is None:
    return
   
Fan module should not proceed furthur.
Please review the current changes

Reply:
Yes, I checked the implementation of `get_sensor_list_by_type`, the returned response should be checked and in case of `None`, fan module should not proceed further. 
Have you implemented this? because I still see the old code.

Reply:
Yes, I have.
check #L248-249

Reply:
@jjxsg  previously I pushed the old code,
Please check the latest commit

Reply:
#L243-244

Reply:
Looks good, ideally there should be a log statement if response is `None ` but that can be added once logging is finalized.

Reply:
@gargpalak048 please sign your commits as well.